packages/compiler-sfc/src/cssVars.ts

```diff
@@ -20,9 +20,12 @@ export function genCssVarsFromList(
   isSSR = false
 ): string {
   return `{\n  ${vars
-    .map(
-      key => `"${isSSR ? `--` : ``}${genVarName(id, key, isProd)}": (${key})`
-    )
+    .map(key => {
+      const varName = genVarName(id, key, isProd)
+      return `"${
+        isSSR ? `--${varName.replace(/\\/, '\\\\')}` : varName
```

Comment:
Try line 38 regex to escape all symbols

Reply:
It has escaped `\` but this only happens in ssr dev model. see https://github.com/vuejs/core/issues/7823#issuecomment-1460230641